posthog/temporal/batch_exports/snowflake_batch_export.py

```diff
@@ -100,6 +100,13 @@ class SnowflakeRetryableConnectionError(Exception):
     pass
 
 
+class SnowflakeTableNotFoundError(Exception):
+    """Raised when a table is not found in Snowflake."""
+
+    def __init__(self, table_name: str):
+        super().__init__(f"Table '{table_name}' not found in Snowflake")
```

Comment:
Given the rest of the code, we would only be raising this if a user happens to delete the Snowflake table after we have created and inserted data, but before we have a chance to merge, correct?

Reply:
Seems like it would be very rare, but good to have this as non-retryable already.

Reply:
yes, it probably won't ever happen, as you say, but just in case